lib/tasks/data_migration.rake

```diff
@@ -82,6 +82,31 @@ namespace :data_migration do
     puts "Number of unprocessed/errored defaulters = #{unprocessed_or_errored_defaulters_count}"
   end
 
+  desc "Set default 'recorded_at' for existing records"
+  task set_default_recorded_at_for_existing_records: :environment do
+    # For BloodPressure records,
+    # we default to the device_created_at
+    blood_pressures = BloodPressure.where(recorded_at: nil)
+
+    blood_pressures.each do |record|
+      record.update_column(:recorded_at, record.device_created_at)
+    end
+
+    puts "Total number of BloodPressure records updated = #{blood_pressures.size}"
+
+    # Patients' recorded_at is set to their earliest
+    # BP's device_created_at if older than theirs
+    patients = Patient.where(recorded_at: nil)
+
+    patients.each do |patient|
+      earliest_blood_pressure = patient.blood_pressures.order(device_created_at: :asc).first
```

Comment:
```suggestion
      earliest_blood_pressure = patient.blood_pressures.order(recorded_at: :asc).first
```

Reply:
at this point, the patients don't have `recorded_at`